src/geometry/ArborX_DetailsAlgorithms.hpp

```diff
@@ -417,48 +406,14 @@ struct intersects<PointTag, SphereTag, Point, Sphere>
   }
 };
 
-template <typename Sphere, typename Point>
-struct centroid<SphereTag, PointTag, Sphere, Point>
-{
-  KOKKOS_FUNCTION static void apply(Sphere const &sphere, Point &c)
-  {
-    static_assert(GeometryTraits::dimension<Point>::value ==
-                  GeometryTraits::dimension<Sphere>::value);
-    c = sphere.centroid();
-  }
-};
-
-// calculate the centroid of a box
-template <typename Box, typename Point>
-struct centroid<BoxTag, PointTag, Box, Point>
-{
-  KOKKOS_FUNCTION static void apply(Box const &box, Point &c)
-  {
-    static_assert(GeometryTraits::dimension<Point>::value ==
-                  GeometryTraits::dimension<Box>::value);
-    constexpr int DIM = GeometryTraits::dimension<Point>::value;
-    for (int d = 0; d < DIM; ++d)
-      c[d] = (box.minCorner()[d] + box.maxCorner()[d]) / 2;
-  }
-};
-
-template <typename Point1, typename Point2>
-struct centroid<PointTag, PointTag, Point1, Point2>
-{
-  KOKKOS_FUNCTION static void apply(Point1 const &point, Point2 &c)
-  {
-    c = point;
-  }
-};
-
 template <typename Point>
-struct returnCentroid<PointTag, Point>
+struct centroid<PointTag, Point>
 {
-  KOKKOS_FUNCTION static auto apply(Point const &point) { return point; }
+  KOKKOS_FUNCTION static auto const &apply(Point const &point) { return point; }
```

Comment:
Why did you change to return a reference?

Reply:
Because I don't see a reason to return it by value. Same thing with the centroid of a sphere. I think it was an oversight.

Reply:
Do you see a reason to return by reference?

Reply:
I assume that it makes the compiler job's easier, and makes the intent more clear (in the sense that don't construct anything within the function, rather return something already existing).